pkg/controllers/disruption/controller.go

```diff
@@ -204,12 +210,9 @@ func (c *Controller) executeCommand(ctx context.Context, m Method, cmd Command,
 	commandID := uuid.NewUUID()
 	log.FromContext(ctx).WithValues("command-id", commandID, "reason", strings.ToLower(string(m.Reason()))).Info(fmt.Sprintf("disrupting nodeclaim(s) via %s", cmd))
 
-	stateNodes := lo.Map(cmd.candidates, func(c *Candidate, _ int) *state.StateNode {
-		return c.StateNode
-	})
 	// Cordon the old nodes before we launch the replacements to prevent new pods from scheduling to the old nodes
-	if err := state.RequireNoScheduleTaint(ctx, c.kubeClient, true, stateNodes...); err != nil {
-		return fmt.Errorf("tainting nodes with %s (command-id: %s), %w", pretty.Taint(v1.DisruptedNoScheduleTaint), commandID, err)
+	if err := c.MarkDisrupted(ctx, m, cmd.candidates...); err != nil {
+		return fmt.Errorf("marking disrupted (command-id: %s), %w", commandID, err)
```

Comment:
I think we probably want to mask any notfound errors here.

Reply:
I masked the errors in `MarkDisrupted`, so this code didn't change.